File: src/Graph/Graph.scala
package graph

class Graph(graphType: Int, vertexList: Array[Int], edgeList: Array[(Int, Int, Int)]){    //edgeList -> (src, dest, weight)
    var adjacencyMatrix: Array[Array[Int]] = null;
    var adjacencyList: Array[List[(Int, Int)]] = null; //dest, weight

    graphType match {
        case 0 => { //adjacency matrix
            //create a 2d array of size vertexList.length x vertexList.length
            var vertexLen = vertexList.length;
            adjacencyMatrix = Array.ofDim[Int](vertexLen, vertexLen);

            //initialize adjacency matrix

            for(i <- 0 until vertexLen){
                for(j <- 0 until vertexLen){
                    println("i: " + i + " j: " + j);
                    //if the i, j in edgelist then set the weight
                    adjacencyMatrix(i)(j) = Int.MaxValue;   //infinity
                }
            }

            //populate adjacency matrix with edgeList
            for(i <- 0 until edgeList.length){
                adjacencyMatrix(edgeList(i)._1)(edgeList(i)._2) = edgeList(i)._3;
                adjacencyMatrix(edgeList(i)._2)(edgeList(i)._1) = edgeList(i)._3;
            }
        }

        case 1 => { //adjacency list
            //create a list of size vertexList.length
            adjacencyList = Array.ofDim[List[(Int, Int)]](vertexList.length);

            //initialize adjacency list
            for(i <- 0 until vertexList.length){
                adjacencyList(i) = List();
            }

            //populate adjacency list with edgeList
            for(i <- 0 until edgeList.length){
                adjacencyList(edgeList(i)._1) = (edgeList(i)._2, edgeList(i)._3) :: adjacencyList(edgeList(i)._1);
                adjacencyList(edgeList(i)._2) = (edgeList(i)._1, edgeList(i)._3) :: adjacencyList(edgeList(i)._2);
            }
        }

        case _ => {
            println("Invalid graph type.");
        }
    }

    def getGraphType() : Int = {
        return graphType;
    }

    def printGraph(): Unit = {
        graphType match{
            case 0 => { //adjacency matrix
                for(i <- 0 until vertexList.length){
                    for(j <- 0 until vertexList.length){
                        print(adjacencyMatrix(i)(j) + " ");
                    }
                    println();
                }
            }

            case 1 => { //adjacency list
                for(i <- 0 until vertexList.length){
                    print(i + ": ");
                    for(j <- 0 until adjacencyList(i).length){
                        print("(" + adjacencyList(i)(j)._1 + "," + adjacencyList(i)(j)._2 + ") ");
                    }
                    println();
                }
            }

            case _ => {
                println("Invalid graph type.");
            }
        }
    }

    def getEdgeCount() : Int = {
        return edgeList.length;
    }

}

File: Python/Graph.py
class Graph:
    def __init__(self, graph_type, vertex_list, edge_list):
        self.adjacency_matrix = None
        self.adjacency_list = None  # dest, weight

        if graph_type == 0:  # adjacency matrix
            # create a 2D list of size len(vertex_list) x len(vertex_list)
            vertex_len = len(vertex_list)
            self.adjacency_matrix = [[float('inf')] * vertex_len for _ in range(vertex_len)]

            # initialize adjacency matrix
            #for i in range(vertex_len):
                ##for j in range(vertex_len):
                    ##self.adjacency_matrix[i][j] = float('inf')

            # populate adjacency matrix with edge_list
            for i in range(len(edge_list)):
                self.adjacency_matrix[edge_list[i][0]][edge_list[i][1]] = edge_list[i][2]
                self.adjacency_matrix[edge_list[i][1]][edge_list[i][0]] = edge_list[i][2]

        elif graph_type == 1:  # adjacency list
            # create a list of size len(vertex_list)
            self.adjacency_list = [[] for _ in range(len(vertex_list))]

            # populate adjacency list with edge_list
            for i in range(len(edge_list)):
                self.adjacency_list[edge_list[i][0]].append((edge_list[i][1], edge_list[i][2]))
                self.adjacency_list[edge_list[i][1]].append((edge_list[i][0], edge_list[i][2]))

        else:
            print("Invalid graph type.")

    def get_graph_type(self):
        return self.graph_type

    def print_graph(self):
        if self.adjacency_matrix is not None:  # adjacency matrix
            for row in self.adjacency_matrix:
                print("\t".join(str(weight) if weight != float('inf') else "-" for weight in row))
        elif self.adjacency_list is not None:  # adjacency list
            for i, adjacency in enumerate(self.adjacency_list):
                print(f"{i}: " + " ".join(f"({dest},{weight})" for dest, weight in adjacency))
        else:
            print("Invalid graph type.")

    def get_edge_count(self):
        return len(self.edge_list)